editor/editor_help.cpp

```diff
@@ -2859,6 +2861,35 @@ String EditorHelp::get_cache_full_path() {
 	return EditorPaths::get_singleton()->get_cache_dir().path_join(vformat("editor_doc_cache-%d.%d.res", VERSION_MAJOR, VERSION_MINOR));
 }
 
+String EditorHelp::get_script_doc_cache_full_path() {
+	return EditorPaths::get_singleton()->get_project_settings_dir().path_join("editor_script_doc_cache.res");
+}
+
+void EditorHelp::add_doc(const DocData::ClassDoc &p_class_doc) {
+	get_doc_data()->add_doc(p_class_doc);
+	if (p_class_doc.is_script_doc) {
+		_delete_script_cache();
+	}
+}
+
+void EditorHelp::remove_doc(const String &p_class_name) {
+	DocTools *dt = get_doc_data();
+	if (dt->has_doc(p_class_name) && dt->class_list[p_class_name].is_script_doc) {
+		_delete_script_cache();
+	}
+	dt->remove_doc(p_class_name);
```

Comment:
```suggestion
	if (dt->has_doc(p_class_name)) {
		if(dt->class_list[p_class_name].is_script_doc) {
			_delete_script_cache();
		}
		dt->remove_doc(p_class_name);
	}
```

Reply:
Your point is taken, but I'm not sure about the suggestion! We only want to delete the script doc cache when the doc being removed is from a script though, right? The `EditorHelp::has/add/remove_doc()` interface is now meant to be used instead of `EditorHelp::get_doc_data()->has/add/remove_doc()`, and it's responsible for keeping cache consistency.

Either way, I moved some things around here.